packages/compass-app-stores/src/stores/instance-store.js

```diff
@@ -251,6 +251,17 @@ store.onActivated = (appRegistry) => {
     store.refreshInstance(appRegistry);
   });
 
+  appRegistry.on('collection-renamed', async ({ from, to }) => {
+    // we must fetch the old collection's metadata before refreshing because refreshing the
+    // collection metadata will remove the old collection from the model.
+    const metadata = await store.fetchCollectionMetadata(from);
+    appRegistry.emit('refresh-collection-tabs', {
+      metadata,
+      newNamespace: to,
+    });
+    appRegistry.emit('refresh-data');
```

Comment:
This seems a bit drastic, we are always renaming in the same database namespace, right? I think `refreshNamespace` should be enough here